.ruff.toml

```diff
@@ -0,0 +1,81 @@
+# Exclude a variety of commonly ignored directories.
+exclude = [
+    ".bzr",
+    ".direnv",
+    ".eggs",
+    ".git",
+    ".git-rewrite",
+    ".hg",
+    ".ipynb_checkpoints",
+    ".mypy_cache",
+    ".nox",
+    ".pants.d",
+    ".pyenv",
+    ".pytest_cache",
+    ".pytype",
+    ".ruff_cache",
+    ".svn",
+    ".tox",
+    ".venv",
+    ".vscode",
+    "__pypackages__",
+    "_build",
+    "buck-out",
+    "build",
+    "dist",
+    "node_modules",
+    "site-packages",
+    "venv",
+    "*.ipynb",  # until we have a reason to include nb formatting
+]
+
+# Same as Black.
+line-length = 100 # up for debate
+indent-width = 4
+
+# Assume minimum Python 3.10
+target-version = "py310"
+
+[lint]
+# Enable Pyflakes (`F`) and a subset of the pycodestyle (`E`)  codes by default.
+select = [
+    "F",  # Pyflakes
+    "I",  # isort
+    "W",  # pycodestyle warnings
+    "E",  # pycodestyle errors
+    # "C901"  # mccabe complexity  TODO: add this back in 
+    "UP",
+]   
+ignore = [
+    "F401",  # unused import - (TODO: fix this when refactoring tests)
+    "F811",  # redefinition (TODO: fix this when refactoring tests)
+    "E402",   # module-level import not at top (conflicts w/ isort)
+    "E501",  # line-length violations (black enforces these)
+    "E731",  # lambda expressions (TODO: we should fix these)
+    "E722",  # bare excepts (TODO: we should fix these in a follow-up PR)
```

Comment:
I plan to fix these in follow-up PRs, because several of them are actual refactoring that I don't want to sneak into this PR.